README.md

```diff
@@ -44,8 +44,13 @@ java -jar zipkin-sparkstreaming-job.jar \
   --zipkin.sparkstreaming.sparkMaster=spark://127.0.0.1:7077
 ```
 
+
 ## Key Components
 
+The image below shows the internal architecture of zipkin spark streaming job. StreamFactory is a extensible interface that ingests data from Kafka or any other transport. The filtering step filters spans based on criteria like service name(#33). The aggregation phase groups the spans by time or trace ID. The adjuster phase is useful for making adjustments to spans that belong to the same trace. For example, the FinagleAdjuster fixes known bugs in the old finagle zipkin tracer. The final consumer stage persists the data to a storage system like ElasticSearch service. 
+
+![zipkinspark_arch](https://cloud.githubusercontent.com/assets/93836/23466070/df14857e-fe4e-11e6-89ad-2294d435b7fa.png)
```

Comment:
another advantage to ascii art is it is trivial to add a phase like filtering when the feature actually exists. For example, instead of mentioning a phase that doesn't exist yet in the top-level readme, then pointing to an issue, we can just add the phase in the diagram when the feature exists.. When we aren't rendering into a png or such, the pressure to make a diagram include future features is less.